src/index.js

```diff
@@ -0,0 +1,195 @@
+};
+
+const landscapes = {
+  coldLandscape: "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲",
+  coolLandscape: "🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃",
+  warmLandscape: "🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷",
+  hotLandscape: "🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂",
+};
+
+const skyOptions = {
+  sunny: "☁️ ☀️ ☁️ ☀️ ☀️  ☁️",
+  cloudy: "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️",
+  rainy: "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧",
+  snowy: "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨",
+};
+
+const landscapeUpdate = () => {
+  let newLandscape;
+  if (state.currentTemperature >= 80) {
+    newLandscape = landscapes.hotLandscape;
+  } else if (state.currentTemperature >= 70) {
+    newLandscape = landscapes.warmLandscape;
+  } else if (state.currentTemperature >= 60) {
+    newLandscape = landscapes.coolLandscape;
+  } else {
+    newLandscape = landscapes.coldLandscape;
+  }
+
+  state.landscapeField.textContent = newLandscape;
+};
+
+//--- wave 5
+const skyOptionsUpdate = () => {
+  let skyValue = state.skyOptions.value;
+  if (skyValue === "2") {
+    state.skyField.textContent = skyOptions.sunny;
+  } else if (skyValue === "3") {
+    state.skyField.textContent = skyOptions.cloudy;
+  } else if (skyValue === "4") {
+    state.skyField.textContent = skyOptions.rainy;
+  } else if (skyValue === "5") {
+    state.skyField.textContent = skyOptions.snowy;
+  } else {
+    state.skyField.textContent = "";
+  }
+};
+
+const weatherOptions = {
+  Clouds: "3",
+  Haze: "3",
+  Mist: "3",
+  Clear: "2",
+  Rain: "4",
+  Drizzle: "4",
+  Thunderstorm: "4",
+  Snow: "5",
+};
+
+const state = {
+  cityTextField: null,
+  headerCity: null,
+  cityResetButton: null,
+  landscapeField: null,
+  skyField: null,
+  skyOptions: null,
+  currTempButton: null,
+  temperatureValue: null,
+  currentTemperature: 50,
+  temperatureIncreaseButton: null,
+  temperatureDecreaseButton: null,
+};
+
+const getTemperatureColor = (temperature) => {
+  if (temperature >= 80) {
+    return "red";
+  } else if (temperature >= 70 && temperature <= 79) {
+    return "orange";
+  } else if (temperature >= 60 && temperature <= 69) {
+    return "yellow";
+  } else if (temperature >= 50 && temperature <= 59) {
+    return "green";
+  } else if (temperature <= 49) {
+    return "teal";
+  }
+};
+
+const tempColorUpdate = () => {
+  state.temperatureValue.className = getTemperatureColor(
+    state.currentTemperature
+  );
+};
+
+const increaseTemperature = () => {
+  state.currentTemperature += 1;
+  state.temperatureValue.textContent = state.currentTemperature;
+  allGraphicsUpdate();
+};
+
+const decreaseTemperature = () => {
+  state.currentTemperature -= 1;
+  state.temperatureValue.textContent = state.currentTemperature;
+  allGraphicsUpdate();
+};
+
+const locUrl = "http://localhost:5000/location";
+const weatherUrl = "http://localhost:5000/weather";
+
+const getTempFromLocation = () => {
+  const q = state.cityTextField.value;
+  axios
+    .get(locUrl, { params: { q } })
+    .then((response) => {
+      const lat = response.data[0].lat;
+      const lon = response.data[0].lon;
+      axios
```

Comment:
Rather than chaining the location call directly into the weather call, consider implementing the API calls a two seperate helper functions: one to get the location given a city, and one to get the weather given a location. Then `getTempFromLocation` could be implemented by calling those two helper functions. We could then make a further function that uses the results of `getTempFromLocation` to update the app data and display.

Try to locate independent pieces of functionality that can be moved into self-contained helpers (not accessing globals or state, just parameters in and a returned result), then implement the overarching behavior using those helpers. This make for more maintainable, self-descriptive code.